bpfassets/perf_event/perf_event.c

```diff
@@ -238,8 +238,17 @@ int kprobe__finish_task_switch(struct pt_regs *ctx, struct task_struct *prev)
     return 0;
 }
 
+struct trace_event_raw_softirq_entry {
+        unsigned short common_type;
+        unsigned char common_flags;
+        unsigned char common_preempt_count;
+        int common_pid;
+        unsigned int vec;
+};
+
 // per https://www.kernel.org/doc/html/latest/core-api/tracepoint.html#c.trace_softirq_entry
-TRACEPOINT_PROBE(irq, softirq_entry)
+// TRACEPOINT_PROBE(irq, softirq_entry)
+int tracepoint__irq__softirq_entry(struct trace_event_raw_softirq_entry *args)
```

Comment:
from https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md
seems TRACEPOINT_PROBE(irq, softirq_entry) is identical to line 251 ? didn't get the reason of this change ..

Reply:
see the error message in #777. This is likely caused by bcc macro
 
```
/virtual/main.c:249:17: error: incomplete definition of type 'struct tracepoint__irq__softirq_entry'
```

Reply:
ok, I didn't see this in my env, not sure it's unique to the env that reported this issue